src/pages/index.js

```diff
@@ -100,30 +156,133 @@ function closeModal(modal) {
   document.removeEventListener("keydown", handleEscape); // remove escape button functionality
 }
 
+// update profile name and description
 function handleProfileFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = nameInput.value;
-  profileJob.textContent = jobInput.value;
-  closeModal(editProfileModal);
+
+  const submitButton = evt.submitter;
+  // Display "Saving..." when request being sent
+  setButtonText(submitButton, true);
+
+  api
+    .editUserInfo({
+      name: profileNameInput.value,
+      about: profileJobInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileJob.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // change back to "Save" once complete
+      setButtonText(submitButton, false);
+    });
+}
+
+// update avatar
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  // Display "Saving..." when request being sent
+  setButtonText(submitButton, true);
+
+  api
+    .editProfilePicture(editAvatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      evt.target.reset(); // reset the form
+      disableButton(submitButton, settings);
+      closeModal(editAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // change back to "Save" once complete
+      setButtonText(submitButton, false);
+    });
 }
 
+// add new card
 function handlePostFormSubmit(evt) {
   evt.preventDefault();
-  const newData = { name: postCaptionInput.value, link: postLinkInput.value };
 
-  // add card to browser
-  const newCardContent = getCardElement(newData);
-  cardsList.prepend(newCardContent);
+  const submitButton = evt.submitter;
+  // Display "Saving..." when request being sent
+  setButtonText(submitButton, true);
+
+  api
+    .addCard({ name: newPostCaptionInput.value, link: newPostLinkInput.value })
+    .then((data) => {
+      const newData = {
+        name: data.name,
+        link: data.link,
+        _id: data._id,
+      };
+
+      // add card to browser
+      const newCardContent = getCardElement(newData);
+      cardsList.append(newCardContent);
+
+      // reset input values
+      evt.target.reset();
+
+      // disable submit button
+      disableButton(newPostSubmitButton, settings);
+
+      closeModal(newPostModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // change back to "Save" once complete
+      setButtonText(submitButton, false);
+    });
+}
+
```

Comment:
* Please, make the `confirmation` modal match the `mobile` design on 320px https://snipboard.io/4QR6Ew.jpg